src/backends/DomRenderer.js

```diff
@@ -71,7 +72,51 @@ export default class DomRenderer {
   attach() {
     // In the case the element node is external and it is already in the DOM.
     if (this.element.parentNode) return
-    this.head.appendChild(this.element)
+
+    let anchorEl = null
+
+    const {index, jss} = this.options
+    const {registry} = jss.sheets
+
+    if (registry.length > 1) {
+      // Try to insert by index if set
+      if (typeof index === 'number') {
+        for (let i = 0; i < registry.length; i++) {
+          const sheet = registry[i]
+          if (
+            !sheet.attached ||
+            !sheet.options.index ||
+            sheet.options.index < index
+          ) continue
+          anchorEl = sheet.renderer.element
+          break
+        }
+      }
+
+      // Otherwise insert after the last attached
+      if (!anchorEl) {
+        for (let i = registry.length - 1; i >= 0; i--) {
+          const registrySheet = registry[i]
+          if (registrySheet.attached) {
+            anchorEl = registrySheet.renderer.element.nextElementSibling
+            break
+          }
+        }
+      }
+    }
+
+    if (!anchorEl) {
+      // Try find a comment placeholder if registry is empty
+      for (let i = 0; i < document.head.childNodes.length; i++) {
+        const el = document.head.childNodes[i]
+        if (el.nodeType === 8 && el.nodeValue === 'jss') {
+          anchorEl = el
+          break
+        }
+      }
+    }
+
+    this.head.insertBefore(this.element, anchorEl)
   }
 
   /**
```

Comment:
I would try to split some of this code into separate functions if it doesn't hurt performance too much, should be optimized through inlining by the runtime.